posts/2021-04-16-mp-context-propagration-21004.adoc

```diff
@@ -0,0 +1,283 @@
+  <file name="/usr/local/postgresql/postgresql-42.2.18.jar"/>
+</library>
+----
+
+Find out more at link:{url-prefix}/docs/latest/reference/config/connectionManager.html[Open Liberty Connection Manager Documentation]
+
+[#SSO]
+== Single Sign-On (SSO) LTPA and JWT support authentication filter>>
+
+With this new enhancement users can now use the authentication filter to select which HTTP servlet request should use LTPA and JWT for the SSO authentications. To find out more about this take a look at link:{url-prefix}/blog/2020/07/02/disable-default-cookies-20007.html[Disabling default LTPA cookies for TAI/SPNEGO or default JWT cookies for JWT SSO on Open Liberty 20.0.0.7]
+
+
+//end::features[]
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21004+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.4].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16113[Shared Class Cache not generated on Windows]
++
+The IBM/OpenJ9 JDK Shared Class Cache may not have been generated on Windows if `IBM_JAVA_OPTIONS/OPENJ9_JAVA_OPTIONS` was not set causing slower startup performance.  With this fix, a Shared Class Cache will be created at `<WLP_USER_DIR>/servers/.classCache` improving the startup performance.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16054[HSTS Header not added on responses with 404 status]
++
+In certain cases where the response is fully handled by the HTTP transport without invoking the WebContainer engine (e.g-a 404 response), the HTTP Strict-Transport-Security (`HSTS`) header was improperly omitted from the response even though the link:{url-prefix}/docs/21.0.0.3/reference/config/webContainer.html[`addStrictTransportSecurityHeader`] was properly configured.  The HTTP transport's parsing of these properties now has new tracing, the HTTP transport will inspect the response and, when configured to do so, add the `HSTS` header if it is missing and if the scheme is 'https'. The `HTTPDispatcher=all` trace level can be used to see what the resulting header value will look like.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15336[Replace DNS lookup with regular expression to get the domain name in SSO Cookie Domain function]
++
+A bug in the SSO Cookie domain function caused an unnecessary DNS lookup when sending a regular HTTP Request to generate LTPA token. To improve performance, a regex will mow be used instead to obtain the domain name.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15989[Bump netty dependencies to 4.1.62.Final]
++
+Although Open Liberty is not vulnerable, we've updated Netty (used by the link:{url-prefix}/docs/21.0.0.3/reference/feature/grpcClient-1.0.html[gRPC Client 1.0] feature) to 4.1.62.Final in order to pull in a fix for a CVE.
```

Comment:
Did this fix not go into 21.0.0.3?

Reply:
My bad this is another version update, should `/docs/21.0.0.3/reference/` -> `/docs/latest/reference/`?

Reply:
Doesn't seem to be in the blog